### PR TITLE
fix #235406: standard keyboard shortcuts missing

### PR DESCRIPTION
**Pull request by aryamanvinchhi**

### CODE REVIEW

File: mscore/shortcut.cpp [hunk 1]
```diff
@@ -3783,6 +3789,7 @@ QString Shortcut::keysToString() const
                   s += "; ";
             s += Shortcut::keySeqToString(_keys[i], QKeySequence::NativeText);
             }
+      s += Shortcut::keySeqToString(_standardKey, QKeySequence::NativeText);
```
Comment (lasconic) [inline mscore/shortcut.cpp]: To be safe, what about guarding this with the following?
```
if (s.isEmpty() && _standardKey != QKeySequence::UnknownKey)
```

Comment (aryamanvinchhi) [inline mscore/shortcut.cpp]: adding this and pushing

**Comment by Jojo-Schmitz** [comment]
When squashing, there is no Need to Keep all commit messages as comments. And the real commit message here should be more verbose, like "fix #235406: standard keyboard shortcuts missing"

**Comment by Jojo-Schmitz** [comment]
This PR seems to also fix #23078, right?

File: mscore/shortcut.cpp [hunk 2]
```diff
@@ -3643,6 +3643,12 @@ void Shortcut::setKeys(const QList<QKeySequence>& ks)
             _action->setShortcuts(_keys);
       }
 
+void Shortcut::setStandardKey(QKeySequence::StandardKey k) {
```
Comment (Jojo-Schmitz) [inline mscore/shortcut.cpp]: Line feed before the `{` I think

Comment (lasconic) [inline mscore/shortcut.cpp]: Please add a comment above the function, just like the one of setKeys

**Comment by aryamanvinchhi** [comment]
Made these changes.

File: mscore/shortcut.cpp [hunk 3]
```diff
@@ -3644,6 +3644,17 @@ void Shortcut::setKeys(const QList<QKeySequence>& ks)
...
 
 //---------------------------------------------------------
+//   setStandardKey
+//---------------------------------------------------------
+
+void Shortcut::setStandardKey(QKeySequence::StandardKey k)
+      {
+      _standardKey = k;
+      if (_action && k != QKeySequence::UnknownKey)
+            _action->setShortcut(_standardKey);
```
Comment (Jojo-Schmitz) [inline mscore/shortcut.cpp]: As mentioned on IRC, it should be `setShortcuts()` here, as with `setShortcut()` only one of possibly multiple shortcuts is set. And only one is shown in Preferences/Shortcuts, e.g. on Windows for File Close only Strg+F4, not also the other Ctrl+W (I guess for Mac it is the other way round). Maybe also another `for` loop is needed in keysToString()?

Comment (lasconic) [inline mscore/shortcut.cpp]: Yes, please, change this to setShortcuts

File: mscore/shortcut.cpp [hunk 4]
```diff
@@ -3783,6 +3794,13 @@ QString Shortcut::keysToString() const
                   s += "; ";
             s += Shortcut::keySeqToString(_keys[i], QKeySequence::NativeText);
             }
+      if (s.isEmpty() && _standardKey != QKeySequence::UnknownKey) {
```
Comment (lasconic) [inline mscore/shortcut.cpp]: Did you test this?

Comment (lasconic) [inline mscore/shortcut.cpp]: I would remove the use of _action and have a look to QKeySequence::keyBindings http://doc.qt.io/qt-5/qkeysequence.html#keyBindings

File: mscore/shortcut.cpp [hunk 5]
```diff
@@ -3783,6 +3794,13 @@ QString Shortcut::keysToString() const
...
+            for (int i = 0; i < QKeySequence::keyBindings(_standardKey).length(); i++) {
+                  if (i)
+                        s += "; ";
+                  s += Shortcut::keySeqToString(QKeySequence::keyBindings(_standardKey)[i], QKeySequence::NativeText );
```
Comment (Jojo-Schmitz) [inline mscore/shortcut.cpp]: picky mode: a space too much before the closing );

**Comment by Jojo-Schmitz** [comment]
I maybe wrong, but I think the 2nd part of this PR might/should go into 2.2 too, while the 1st part doesn't apply there, so it might be a good idea to make it 2 separate commits for easy cherry picking?

**Comment by lasconic** [comment]
No both are fine.

File: mscore/shortcut.cpp [hunk 6]
```diff
@@ -3783,6 +3794,13 @@ QString Shortcut::keysToString() const
                   s += "; ";
             s += Shortcut::keySeqToString(_keys[i], QKeySequence::NativeText);
             }
+      if (s.isEmpty() && _standardKey != QKeySequence::UnknownKey) {
+            for (int i = 0; i < QKeySequence::keyBindings(_standardKey).length(); i++) {
+                  if (i)
+                        s += "; ";
+                  s += Shortcut::keySeqToString(QKeySequence::keyBindings(_standardKey)[i], QKeySequence::NativeText);
```
Comment (lasconic) [inline mscore/shortcut.cpp]: Could you use a variable instead of calling twice the keyBindings function ?

**Pull request merged**